fms/models/gpt_oss.py

```diff
@@ -0,0 +1,667 @@
+from dataclasses import dataclass
+from typing import Any, Mapping, Optional, Tuple
+import re
+import math
+
+from typing_extensions import Unpack
+
+import torch
+import torch.nn as nn
+
+from fms import models
+from fms.distributed.strategy import DistributedStrategy, NoOpStrategy
+from fms.modules.attention import (
+    AttentionKwargs,
+    MultiHeadAttention,
+    get_attention_type,
+)
+from fms.utils import serialization
+from fms.utils.config import ModelConfig
+
+from fms.modules.feedforward import MOEFeedForward
+from fms.modules.positions import RotaryEmbedding
+
+from fms.modules.layernorm import LayerNormParameterized
+
+FP4_VALUES = [
+    +0.0,
+    +0.5,
+    +1.0,
+    +1.5,
+    +2.0,
+    +3.0,
+    +4.0,
+    +6.0,
+    -0.0,
+    -0.5,
+    -1.0,
+    -1.5,
+    -2.0,
+    -3.0,
+    -4.0,
+    -6.0,
+]
+
+
+@dataclass
+class GptOssConfig(ModelConfig):
+    num_experts: int = 32
+    src_vocab_size: int = 201088
+    emb_dim: int = 2880
+    head_dim: int = 64
+    num_attention_heads: int = 64
+    sliding_window: int = 128
+    rope_base: float = 150000.0
+    tie_heads = False
+    activation_fn: str = "silu"
+    initializer_range: float = 0.02
+    max_expected_seq_len = 131072
+    top_k_experts = 4
+    router_aux_loss_coef: float = 0.9
+    output_router_logits = False
+    layer_types = None
+    pad_id: int = -1
+    nheads: int = 64
+    nlayers: int = 24
+    norm_eps: float = 1e-05
+    kvheads: int = 8
+    p_dropout: float = 0.0
+    fused_weights: bool = True
+    linear_config: Optional[Mapping[str, Any]] = None
+    hidden_grow_factor: float = 1.0
+    multiple_of: int = 256
+    embedding_multiplier: float = 1.0
+    residual_multiplier: float = 1.0
+    logits_scaling: float = 1.0
+    attention_multiplier: float = 1.0
```

Comment:
I'm not seeing any of these multipliers in HF or the GPT OSS repo. Can we get rid of them?